plugin-k8s/src/lib.rs

```diff
@@ -50,25 +54,45 @@ impl AlumetPlugin for K8sPlugin {
 
     fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
         let config = deserialize_config(config).context("invalid config")?;
-        Ok(Box::new(K8sPlugin { config: config, watcher: None, metrics: None }))
+        Ok(Box::new(K8sPlugin {
+            config: config,
+            watcher: None,
+            metrics: None,
+        }))
     }
 
     fn start(&mut self, alumet: &mut alumet::plugin::AlumetStart) -> anyhow::Result<()> {
         let v2_used: bool = cgroup_v2::is_accessible_dir(&PathBuf::from("/sys/fs/cgroup/"));
         if !v2_used {
             anyhow::bail!("Cgroups v2 are not being used!");
         }
-        let metrics_result = Metrics::new(alumet);
-        let metrics = metrics_result?;
-        self.metrics = Some(metrics.clone());
-        let final_li_metric_file: Vec<CgroupV2MetricFile> = cgroup_v2::list_all_k8s_pods_file(&self.config.path)?;
+        self.metrics = Some(Metrics::new(alumet)?);
+
+        let hostname_ostring = gethostname();
+        let hostname = hostname_ostring
+            .to_str()
+            .context("Invalid UTF-8 in Hostname")?
+            .to_string();
+
+        let final_list_metric_file: Vec<CgroupV2MetricFile> = cgroup_v2::list_all_k8s_pods_file(
+            &self.config.path,
+            hostname.clone(),
+            self.config.kubernetes_api_url.clone(),
+        )?;
+        self.config.hostname = hostname;
```

Comment:
Does that mean that we could set the hostname in the config file ?

Reply:
yes we can

Reply:
But aren't you overwriting it with this line ?